add Debian packaging

tomkralidis opened the pull request.

neishm commented:
Thanks for the patch!  I have a couple of notes/questions:
1) The package is already released under LGPL-3 (as defined in the `COPYING` and `COPYING.lesser` files), so the `LICENSE` file for GPL-3 may add confusion?  Don't think it's needed in this case.
2) In `debian/copyright`, the license should be LGPL-3, and I don't think there should be text about "MapServer" in there?
3) Probably the control file would need to have a dependency on rpnpy, librmn, and libvgrid in order to function? (and none of those packages are Debianized as far as I'm aware?)
4) Is it standard policy for upstream to include Debian packaging information in an upstream repository?  It's been a long time since I touched Debian packaging, but I thought there was a policy of having a separation of concerns between upstream and Debian maintainers?  I.e., someone else (not upstream) would manage the Debian packaging, pulling the pristine upstream source and applying patches as they see fit?  If the packaging stuff resides in this repo, then it would be nice to have some long-term support for maintaining the files, since I have no idea how to keep that mechanism up-to-date!

tomkralidis commented:
> Thanks for the patch! I have a couple of notes/questions:
> 
>     1. The package is already released under LGPL-3 (as defined in the `COPYING` and `COPYING.lesser` files), so the `LICENSE` file for GPL-3 may add confusion?  Don't think it's needed in this case.
> 

Fixed.
>     2. In `debian/copyright`, the license should be LGPL-3, and I don't think there should be text about "MapServer" in there?

Fixed.

> 
>     3. Probably the control file would need to have a dependency on rpnpy, librmn, and libvgrid in order to function? (and none of those packages are Debianized as far as I'm aware?)
> 

Ah, I was going off of `setup.py` `install_requires`.  So yes, all dependencies (and their own upstream depedencies) need to be Debianized (with the current primary target being Ubuntu focal).  In this case I suspect it will take some time to harden the dependency tree in the Debian context.  Still, this PR is meant as a first pass.

>     4. Is it standard policy for upstream to include Debian packaging information in an upstream repository?  It's been a long time since I touched Debian packaging, but I thought there was a policy of having a separation of concerns between upstream and Debian maintainers?  I.e., someone else (not upstream) would manage the Debian packaging, pulling the pristine upstream source and applying patches as they see fit?  If the packaging stuff resides in this repo, then it would be nice to have some long-term support for maintaining the files, since I have no idea how to keep that mechanism up-to-date!

I think Sebastien Fortier would help with packaging maintenance?  We can clarify.

Having said this, the scope of this PR is a first pass to allow for refinement.  The target would initially be internal PPAs used for MSC GeoMet.  We can think about broader distribution once the packaging is locked down / works / is mature.

neishm commented:
Thanks!  I'll merge in what's done so far, assuming it actually builds a .deb file :)
For the **fstd2nc-deps** stuff (which includes a modified rpnpy, librmn, and libvgrid), that's currently being constructed in the **python-rpn-wheel** package, in the "fstd2nc-deps" branch:
https://github.com/neishm/python-rpn-wheel/tree/fstd2nc-deps
There's a lot of moving parts in there, but you can get a self-contained source by running:
```
make sdist
```
This will put a `.zip` file in the `wheelhouse/` directory.  If you unzip that, it should give you a clean source for trying out a `debian/` subdirectory structure.
